staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -152,6 +154,9 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	var httpCode int
+	defer handlers.RecordMetrics(w, req, requestInfo, httpCode, reqStart)
```

Comment:
Do we really want to add this code to each and every handler? Seems like it would make sense to wrap each handler in a handler that records this.

Reply:
You probably overestimate the number of handlers. We have the proxy handler, the crd handler and basically all other API handlers (impemented in a generic way). 

Though having this as a decorator would be better. Not sure that's feasible code-wise?

Reply:
I see, then I'm fine with this, I just thought we're doing this for every API avilable.